### PR TITLE
docstr/pylint of GP Tuner & CurveFitting Assessor

### PR DESCRIPTION
**Pull request by suiguoxin**

### CODE REVIEW

File: src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py [hunk 1]
```diff
@@ -105,7 +105,6 @@ def assess_trial(self, trial_job_id, trial_history):
         Exception
             unrecognize exception in curvefitting_assessor
         """
-        trial_job_id = trial_job_id
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py]: why delete it? unused variable?

Comment (suiguoxin) [reply src/sdk/pynni/nni/curvefitting_assessor/curvefitting_assessor.py]: > why delete it? unused variable?

There is a "self-assigning-variable" warning here found by pylint.  It's true that this line has no influence.

File: src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py [hunk 2]
```diff
@@ -100,6 +102,7 @@ def logx_linear(x, a, b):
     float
         a * np.log(x) + b
     """
+    # pylint: disable=assignment-from-no-return
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/curvefitting_assessor/curvefunctions.py]: This is not needed.

File: src/sdk/pynni/nni/curvefitting_assessor/model_factory.py [hunk 3]
```diff
@@ -308,6 +308,7 @@ def mcmc_sampling(self):
             new_values = np.random.randn(NUM_OF_INSTANCE, self.effective_model_num) * STEP_SIZE + self.weight_samples
             new_values = self.normalize_weight(new_values)
             # compute alpha(i, j) = min{1, P(j)Q(j, i)/P(i)Q(i, j)}
+            # pylint: disable=assignment-from-no-return
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/curvefitting_assessor/model_factory.py]: Not needed. Use latest pylintrc.

File: src/sdk/pynni/nni/gp_tuner/gp_tuner.py [hunk 4]
```diff
@@ -17,14 +17,17 @@
 # NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
 # DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-'''
-gp_tuner.py
-'''
+"""
+GPTuner is a Bayesian Optimization method where Gaussian Process is used for modeling loss functions.
+
+See : class:`GPTuner` for details.
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/gp_tuner.py]: Space after colon.

File: src/sdk/pynni/nni/gp_tuner/gp_tuner.py [hunk 5]
```diff
@@ -17,14 +17,17 @@
...
+"""
 
 import warnings
 import logging
 import numpy as np
 
+# pylint: disable=no-name-in-module, import-error
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/gp_tuner.py]: Not needed.

File: src/sdk/pynni/nni/gp_tuner/gp_tuner.py [hunk 6]
```diff
@@ -38,9 +41,31 @@
...
 class GPTuner(Tuner):
-    '''
-    GPTuner
-    '''
+    """
+    GPTuner is a Bayesian Optimization method where Gaussian Process is used for modeling loss functions.
+
+    Parameters
+    ----------
+    optimize_mode: str
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/gp_tuner.py]: There should be a space before colon here. My example was incorrect.

File: src/sdk/pynni/nni/gp_tuner/gp_tuner.py [hunk 7]
```diff
@@ -38,9 +41,31 @@
...
+        optimize mode, 'maximize' or 'minimize', by default 'maximize'
+    utility: str
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/gp_tuner.py]: Documented attributes are public APIs. Use underscore name for internal states.

File: src/sdk/pynni/nni/gp_tuner/gp_tuner.py [hunk 8]
```diff
@@ -75,27 +100,20 @@ def __init__(self, optimize_mode="maximize", utility='ei', kappa=5, xi=0, nu=2.5
...
 
     def update_search_space(self, search_space):
-        """Update the self.bounds and self.types by the search_space.json
+        """
+        Update the self.bounds and self.types by the search_space.json file.
 
-        Parameters
-        ----------
-        search_space : dict
+        Override of the abstract method in class:`Tuner`.
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/gp_tuner.py]: `` class:`~nni.tuner.Tuner` ``

File: src/sdk/pynni/nni/gp_tuner/target_space.py [hunk 9]
```diff
@@ -17,39 +17,51 @@
...
+    Transform list params to tuple format. Ensure that an point is hashable by a python dict.
+
+    Parameters
+    ----------
+    params: numpy array
+        array format of parameters
+
+    Returns
+    -------
+    result: tuple
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/target_space.py]: Numpy "returns" does not include a name.

File: src/sdk/pynni/nni/gp_tuner/gp_tuner.py (hunk 5 above)
Comment (suiguoxin) [reply src/sdk/pynni/nni/gp_tuner/gp_tuner.py]: why ? This is not included in pylintrc

**Comment by suiguoxin** [comment]
Updates:
- Updated according to your comments
- refactor of medianstop assessor docstr

File: src/sdk/pynni/nni/gp_tuner/util.py [hunk 10]
```diff
@@ -28,9 +28,21 @@
...
 def _match_val_type(vals, bounds):
-    '''
-    Update values in the array, to match their corresponding type
-    '''
+    """
+    Update values in the array, to match their corresponding type, make sure the value is legal.
+
+    Parameters
+    ----------
+    vals :numpy array
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/util.py]: Space after colon

File: src/sdk/pynni/nni/gp_tuner/util.py [hunk 11]
```diff
@@ -52,32 +64,33 @@ def acq_max(f_acq, gp, y_max, bounds, space, num_warmup, num_starting_points):
...
     It uses a combination of random sampling (cheap) and the 'L-BFGS-B'
-    optimization method. First by sampling `n_warmup` (1e5) points at random,
-    and then running L-BFGS-B from `n_iter` (250) random starting points.
+    optimization method. First by sampling `num_warmup` points at random,
+    and then running L-BFGS-B from `num_starting_points` random starting points.
 
     Parameters
     ----------
-    :param f_acq:
+    param f_acq : UtilityFunction.utility
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/util.py]: No `param`

File: src/sdk/pynni/nni/gp_tuner/util.py [hunk 12]
```diff
@@ -117,36 +130,70 @@ def acq_max(f_acq, gp, y_max, bounds, space, num_warmup, num_starting_points):
...
-        if self.kind == 'ei':
-            return self._ei(x, gp, y_max, self.xi)
-        if self.kind == 'poi':
-            return self._poi(x, gp, y_max, self.xi)
+        """
+        return utility function
+
+        Parameters
+        ----------
+        x: numpy array
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/util.py]: Space

File: src/sdk/pynni/nni/gp_tuner/util.py [hunk 13]
```diff
@@ -52,32 +64,33 @@ def acq_max(f_acq, gp, y_max, bounds, space, num_warmup, num_starting_points):
     A function to find the maximum of the acquisition function
 
     It uses a combination of random sampling (cheap) and the 'L-BFGS-B'
-    optimization method. First by sampling `n_warmup` (1e5) points at random,
-    and then running L-BFGS-B from `n_iter` (250) random starting points.
+    optimization method. First by sampling `num_warmup` points at random,
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/util.py]: Please check if single quote works here

File: src/sdk/pynni/nni/gp_tuner/util.py [hunk 14]
```diff
@@ -117,36 +130,70 @@ def acq_max(f_acq, gp, y_max, bounds, space, num_warmup, num_starting_points):
...
+        if self._kind == 'ei':
+            return self._ei(x, gp, y_max, self._xi)
+        if self._kind == 'poi':
+            return self._poi(x, gp, y_max, self._xi)
         return None
 
     @staticmethod
     def _ucb(x, gp, kappa):
+        """
+        Upper Confidence Bound(UCB) utility function
```
Comment (liuzhe-lz) [inline src/sdk/pynni/nni/gp_tuner/util.py]: Should have a space before `(`

**Pull request merged**